vowpalwabbit/debug_log.h

```diff
@@ -2,9 +2,50 @@
 // individual contributors. All rights reserved. Released under a BSD (revised)
 // license as described in the file LICENSE.
 
+#include "future_compat.h"
 #pragma once
 
-#define VW_DEBUG_ENABLE(x) static bool VW_DEBUG_LOG = x;
+struct vw_dbg
+{
+  static constexpr bool default_log         = false;
+  static constexpr bool learner             = false;
+  static constexpr bool search              = false;
+  static constexpr bool gd                  = false;
```

Comment:
is this a list of reductions that currently have vw_dbg logging inside of it, and if there's a reduction name missing from here what is the meaning you can infer from it?

Reply:
Any trace statements in other reductions will be enabled/disabled by default_log.